---
title: "PhysicoChemicalPropertiesProtein: Calculation of physico-chemical properties of amino acid and protein sequences"
author:
 - name: Thomas Naake
   mail: thomas.naake@embl.de, thomasnaake@googlemail.com
   affiliation: European Molecular Biology Laboratory, Meyerhofstrasse 1, 69117 Heidelberg
package: PhysicoChemicalPropertiesProtein
      
output:  
    BiocStyle::html_document:
        toc_float: true
bibliography: PhysicoChemicalPropertiesProtein-citations.bib
vignette: >
    %\VignetteIndexEntry{Physico-chemical properties of amino acid and protein sequences}
    %\VignetteEngine{knitr::rmarkdown}
    %\VignetteKeywords{Proteomics, Mass Spectrometry, MS, QC}
    %\VignettePackage{PhysicoChemicalPropertiesProtein-vignette}
    %\VignetteEncoding{UTF-8}
---

```{r style, echo = FALSE, results = 'asis'}
BiocStyle::markdown()
```

```{r env, include=FALSE, echo=FALSE, cache=FALSE}
library("knitr")
opts_chunk$set(stop_on_error = 1L)
suppressPackageStartupMessages(library("PhysicoChemicalPropertiesProtein"))
```

# Introduction {#sec-intro}


# Installation

To install this package, start `R` and enter:

```{r, eval = FALSE}
library(devtools)
install_github("tnaake/PhysicoChemicalPropertiesProtein")
```

This will install this package and all eventually missing dependencies.

# Questions and bugs {-}

`PhysicoChemicalPropertiesProtein` is currently under active development. 
If you discover any bugs, typos or develop ideas of improving 
`PhysicoChemicalPropertiesProtein` feel free to raise an issue via
[GitHub](https://github.com/tnaake/PhysicoChemicalPropertiesProtein) or 
send a mail to the developer.

# Calculate physico-chemical properties of amino acid sequences

```{r load_PhysicoChemicalPropertiesProtein}
library("Biostrings")
library("PhysicoChemicalPropertiesProtein")
```

## GRAVY Score

The function `calculateGravyScore` calculates the GRAVY (grand average 
of hydropathy) value of a given amino acid/protein sequence. 
The GRAVY value is calculated by adding the hydropathy value for each 
residue and dividing by the length of the sequence.

The hydropathy values are taken from Table 2 in @Kyte1982.

```{r gravy, eval = TRUE, echo = TRUE}
aa <- AAString("TEST")
calculateGravyScore(aa = aa)
```

## Isoelectric point

The function `calculateIsoelectricPoint` calculates the isoelectric point of a 
given amino acid/protein sequence. Several methods are implemented that use 
different pKa values for the ionizable groups of proteins (N terminal, 
C terminal, and amino acids C, D, E, H, K, R, and Y).

The pKa values are taken from Table 4 in @Kozlowski2016.
 
For polypeptides, the isoelectric point depends primarily on the dissociation 
constants (pKa) for the ionizable groups of seven charged amino acids: 
glutamate (\\delta-carboxyl group), aspartate (\\beta-carboxyl group), cysteine 
(thiol group), tyrosine (phenol group), histidine (imidazole side chains), 
lysine (\\epsilon-ammonium group) and arginine (guanidinium group). Moreover, 
the charge of the terminal groups (NH2 and COOH) can greatly affect the pI 
of short peptides. Generally, the Glu, Asp, Cys, and Tyr ionizable groups 
are uncharged below their pKa and negatively charged above their pKa. 
Similarly, the His, Lys, and Arg ionizable groups are positively charged 
below their pKa and uncharged above their pKa.
 
The implemented algorithm (bisection algorithm) is as in @Kozlowski2016.

```{r IE_point, eval = TRUE, echo = TRUE}
aa <- AAString("TEST")
calculateIsoelectricPoint(aa = aa, method = "EMBOSS")
```

# Appendix {-}

## Session information {-}

All software and respective versions to build this vignette are listed here:

```{r session,eval=TRUE, echo=FALSE}
sessionInfo()
```

## References

